posthog/hogql/functions/mapping.py

```diff
@@ -2045,3 +2098,46 @@ def find_hogql_posthog_function(name: str) -> Optional[HogQLFunctionMeta]:
 def is_allowed_parametric_function(name: str) -> bool:
     # No case-insensitivity for parametric functions
     return name in HOGQL_PERMITTED_PARAMETRIC_FUNCTIONS
+
+
+def get_correct_function_name(name: str) -> str:
+    if HOGQL_CLICKHOUSE_FUNCTIONS.get(name) or HOGQL_AGGREGATIONS.get(name) or HOGQL_POSTHOG_FUNCTIONS.get(name):
+        return name
+
+    name_lower = name.lower()
+    for key in HOGQL_CLICKHOUSE_FUNCTIONS:
+        if key.lower() == name_lower:
+            return key
+
+    for key in HOGQL_AGGREGATIONS:
+        if key.lower() == name_lower:
+            return key
+
+    for key in HOGQL_POSTHOG_FUNCTIONS:
+        if key.lower() == name_lower:
+            return key
+
+    return name
+
+
+def normalize_hogql_function_names(query: str) -> str:
```

Comment:
We should be passing a query around internall in `posthog/hogql` as a `ast.SelectQuery | ast.SelectSetQuery` type please